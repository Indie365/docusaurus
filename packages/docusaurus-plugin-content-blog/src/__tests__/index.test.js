/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import pluginContentBlog from '../index';

describe('loadBlog', () => {
  test('simple website', async () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'website');
    const siteConfig = {
      title: 'Hello',
      baseUrl: '/',
      url: 'https://docusaurus.io',
    };
    const pluginPath = 'blog';
    const plugin = pluginContentBlog(
      {
        siteDir,
        siteConfig,
      },
      {
        path: 'blog',
      },
    );
    const {blogPosts} = await plugin.loadContent();

    expect(blogPosts[0].metadata).toEqual({
      permalink: '/blog/2019/01/01/date-matter',
      source: path.join('@site', pluginPath, 'date-matter.md'),
      title: 'date-matter',
      description: `date inside front matter`,
      date: new Date('2019-01-01'),
      tags: [],
    });
  });
});
