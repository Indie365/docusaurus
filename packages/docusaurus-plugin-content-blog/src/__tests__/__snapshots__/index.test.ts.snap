// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog plugin process blog posts load content 1`] = `
{
  "/blog/tags/tag-1": {
    "items": [
      "/simple/slug/another",
      "/another/tags",
      "/another/tags2",
    ],
    "label": "tag1",
    "pages": [
      {
        "items": [
          "/simple/slug/another",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": "/blog/tags/tag-1/page/2",
          "page": 1,
          "permalink": "/blog/tags/tag-1",
          "postsPerPage": 1,
          "previousPage": undefined,
          "totalCount": 3,
          "totalPages": 3,
        },
      },
      {
        "items": [
          "/another/tags",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": "/blog/tags/tag-1/page/3",
          "page": 2,
          "permalink": "/blog/tags/tag-1/page/2",
          "postsPerPage": 1,
          "previousPage": "/blog/tags/tag-1",
          "totalCount": 3,
          "totalPages": 3,
        },
      },
      {
        "items": [
          "/another/tags2",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 3,
          "permalink": "/blog/tags/tag-1/page/3",
          "postsPerPage": 1,
          "previousPage": "/blog/tags/tag-1/page/2",
          "totalCount": 3,
          "totalPages": 3,
        },
      },
    ],
    "permalink": "/blog/tags/tag-1",
    "unlisted": false,
  },
  "/blog/tags/tag-2": {
    "items": [
      "/another/tags",
      "/another/tags2",
    ],
    "label": "tag2",
    "pages": [
      {
        "items": [
          "/another/tags",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": "/blog/tags/tag-2/page/2",
          "page": 1,
          "permalink": "/blog/tags/tag-2",
          "postsPerPage": 1,
          "previousPage": undefined,
          "totalCount": 2,
          "totalPages": 2,
        },
      },
      {
        "items": [
          "/another/tags2",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 2,
          "permalink": "/blog/tags/tag-2/page/2",
          "postsPerPage": 1,
          "previousPage": "/blog/tags/tag-2",
          "totalCount": 2,
          "totalPages": 2,
        },
      },
    ],
    "permalink": "/blog/tags/tag-2",
    "unlisted": false,
  },
}
`;

exports[`blog plugin process blog posts load content 2`] = `
[
  {
    "content": "simple url slug",
    "id": "/simple/slug/another",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "Sébastien Lorber",
          "title": "Docusaurus maintainer",
          "url": "https://sebastienlorber.com",
        },
      ],
      "date": 2020-08-15T00:00:00.000Z,
      "description": "simple url slug",
      "editUrl": "https://baseEditUrl.com/edit/blog/another-simple-slug-with-tags.md",
      "frontMatter": {
        "author": "Sébastien Lorber",
        "author_title": "Docusaurus maintainer",
        "author_url": "https://sebastienlorber.com",
        "date": 2020-08-15T00:00:00.000Z,
        "slug": "/simple/slug/another",
        "tags": [
          "tag1",
        ],
        "title": "Another Simple Slug",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/another/tags",
        "title": "Another With Tag",
      },
      "permalink": "/blog/simple/slug/another",
      "readingTime": 0.015,
      "source": "@site/blog/another-simple-slug-with-tags.md",
      "tags": [
        {
          "label": "tag1",
          "permalink": "/blog/tags/tag-1",
        },
      ],
      "title": "Another Simple Slug",
      "unlisted": false,
    },
  },
  {
    "content": "with tag",
    "id": "/another/tags",
    "metadata": {
      "authors": [],
      "date": 2020-08-15T00:00:00.000Z,
      "description": "with tag",
      "editUrl": "https://baseEditUrl.com/edit/blog/another-with-tags.md",
      "frontMatter": {
        "date": 2020-08-15T00:00:00.000Z,
        "slug": "/another/tags",
        "tags": [
          "tag1",
          "tag2",
        ],
        "title": "Another With Tag",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/another/tags2",
        "title": "Another With Tag",
      },
      "permalink": "/blog/another/tags",
      "prevItem": {
        "permalink": "/blog/simple/slug/another",
        "title": "Another Simple Slug",
      },
      "readingTime": 0.01,
      "source": "@site/blog/another-with-tags.md",
      "tags": [
        {
          "label": "tag1",
          "permalink": "/blog/tags/tag-1",
        },
        {
          "label": "tag2",
          "permalink": "/blog/tags/tag-2",
        },
      ],
      "title": "Another With Tag",
      "unlisted": false,
    },
  },
  {
    "content": "with tag",
    "id": "/another/tags2",
    "metadata": {
      "authors": [],
      "date": 2020-08-15T00:00:00.000Z,
      "description": "with tag",
      "editUrl": "https://baseEditUrl.com/edit/blog/another-with-tags2.md",
      "frontMatter": {
        "date": 2020-08-15T00:00:00.000Z,
        "slug": "/another/tags2",
        "tags": [
          "tag1",
          "tag2",
        ],
        "title": "Another With Tag",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "permalink": "/blog/another/tags2",
      "prevItem": {
        "permalink": "/blog/another/tags",
        "title": "Another With Tag",
      },
      "readingTime": 0.01,
      "source": "@site/blog/another-with-tags2.md",
      "tags": [
        {
          "label": "tag1",
          "permalink": "/blog/tags/tag-1",
        },
        {
          "label": "tag2",
          "permalink": "/blog/tags/tag-2",
        },
      ],
      "title": "Another With Tag",
      "unlisted": false,
    },
  },
]
`;

exports[`blog plugin works on blog tags without pagination 1`] = `
{
  "/blog/tags/tag-1": {
    "items": [
      "/simple/slug/another",
      "/another/tags",
      "/another/tags2",
    ],
    "label": "tag1",
    "pages": [
      {
        "items": [
          "/simple/slug/another",
          "/another/tags",
          "/another/tags2",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 1,
          "permalink": "/blog/tags/tag-1",
          "postsPerPage": 3,
          "previousPage": undefined,
          "totalCount": 3,
          "totalPages": 1,
        },
      },
    ],
    "permalink": "/blog/tags/tag-1",
    "unlisted": false,
  },
  "/blog/tags/tag-2": {
    "items": [
      "/another/tags",
      "/another/tags2",
    ],
    "label": "tag2",
    "pages": [
      {
        "items": [
          "/another/tags",
          "/another/tags2",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 1,
          "permalink": "/blog/tags/tag-2",
          "postsPerPage": 2,
          "previousPage": undefined,
          "totalCount": 2,
          "totalPages": 1,
        },
      },
    ],
    "permalink": "/blog/tags/tag-2",
    "unlisted": false,
  },
  "/blog/tags/unlisted": {
    "items": [
      "/another/blog-with-tags-unlisted",
    ],
    "label": "unlisted",
    "pages": [
      {
        "items": [
          "/another/blog-with-tags-unlisted",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 1,
          "permalink": "/blog/tags/unlisted",
          "postsPerPage": 1,
          "previousPage": undefined,
          "totalCount": 1,
          "totalPages": 1,
        },
      },
    ],
    "permalink": "/blog/tags/unlisted",
    "unlisted": false,
  },
}
`;

exports[`blog plugin works with blog tags 1`] = `
{
  "/blog/tags/tag-1": {
    "items": [
      "/simple/slug/another",
      "/another/tags",
      "/another/tags2",
    ],
    "label": "tag1",
    "pages": [
      {
        "items": [
          "/simple/slug/another",
          "/another/tags",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": "/blog/tags/tag-1/page/2",
          "page": 1,
          "permalink": "/blog/tags/tag-1",
          "postsPerPage": 2,
          "previousPage": undefined,
          "totalCount": 3,
          "totalPages": 2,
        },
      },
      {
        "items": [
          "/another/tags2",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 2,
          "permalink": "/blog/tags/tag-1/page/2",
          "postsPerPage": 2,
          "previousPage": "/blog/tags/tag-1",
          "totalCount": 3,
          "totalPages": 2,
        },
      },
    ],
    "permalink": "/blog/tags/tag-1",
    "unlisted": false,
  },
  "/blog/tags/tag-2": {
    "items": [
      "/another/tags",
      "/another/tags2",
    ],
    "label": "tag2",
    "pages": [
      {
        "items": [
          "/another/tags",
          "/another/tags2",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 1,
          "permalink": "/blog/tags/tag-2",
          "postsPerPage": 2,
          "previousPage": undefined,
          "totalCount": 2,
          "totalPages": 1,
        },
      },
    ],
    "permalink": "/blog/tags/tag-2",
    "unlisted": false,
  },
  "/blog/tags/unlisted": {
    "items": [
      "/another/blog-with-tags-unlisted",
    ],
    "label": "unlisted",
    "pages": [
      {
        "items": [
          "/another/blog-with-tags-unlisted",
        ],
        "metadata": {
          "blogDescription": "Blog",
          "blogTitle": "Blog",
          "nextPage": undefined,
          "page": 1,
          "permalink": "/blog/tags/unlisted",
          "postsPerPage": 2,
          "previousPage": undefined,
          "totalCount": 1,
          "totalPages": 1,
        },
      },
    ],
    "permalink": "/blog/tags/unlisted",
    "unlisted": false,
  },
}
`;

exports[`last update author and time 1`] = `
[
  {
    "content": "author",
    "id": "author",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "author",
      "editUrl": "https://baseEditUrl.com/edit/blog/author.md",
      "frontMatter": {
        "author": "ozaki",
        "last_update": {
          "author": "seb",
        },
        "slug": "author",
        "title": "Author",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1539502055,
      "lastUpdatedBy": "seb",
      "nextItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "permalink": "/blog/author",
      "readingTime": 0.005,
      "source": "@site/blog/author.md",
      "tags": [],
      "title": "Author",
      "unlisted": false,
    },
  },
  {
    "content": "nothing",
    "id": "nothing",
    "metadata": {
      "authors": [],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "nothing",
      "editUrl": "https://baseEditUrl.com/edit/blog/nothing.md",
      "frontMatter": {
        "slug": "nothing",
        "title": "Nothing",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1539502055,
      "lastUpdatedBy": "Author",
      "nextItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "permalink": "/blog/nothing",
      "prevItem": {
        "permalink": "/blog/author",
        "title": "Author",
      },
      "readingTime": 0.005,
      "source": "@site/blog/nothing.md",
      "tags": [],
      "title": "Nothing",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "both",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/both.md",
      "frontMatter": {
        "author": "ozaki",
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "author": "seb",
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "both",
        "title": "Both",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1609459200,
      "lastUpdatedBy": "seb",
      "nextItem": {
        "permalink": "/blog/lastUpdateDate",
        "title": "Last update date",
      },
      "permalink": "/blog/both",
      "prevItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "readingTime": 0.015,
      "source": "@site/blog/both.md",
      "tags": [],
      "title": "Both",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "lastUpdateDate",
    "metadata": {
      "authors": [],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/lastUpdateDate.md",
      "frontMatter": {
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "lastUpdateDate",
        "title": "Last update date",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1609459200,
      "lastUpdatedBy": "Author",
      "permalink": "/blog/lastUpdateDate",
      "prevItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "readingTime": 0.015,
      "source": "@site/blog/lastUpdateDate.md",
      "tags": [],
      "title": "Last update date",
      "unlisted": false,
    },
  },
]
`;

exports[`last update author only 1`] = `
[
  {
    "content": "author",
    "id": "author",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "author",
      "editUrl": "https://baseEditUrl.com/edit/blog/author.md",
      "frontMatter": {
        "author": "ozaki",
        "last_update": {
          "author": "seb",
        },
        "slug": "author",
        "title": "Author",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1539502055,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "permalink": "/blog/author",
      "readingTime": 0.005,
      "source": "@site/blog/author.md",
      "tags": [],
      "title": "Author",
      "unlisted": false,
    },
  },
  {
    "content": "nothing",
    "id": "nothing",
    "metadata": {
      "authors": [],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "nothing",
      "editUrl": "https://baseEditUrl.com/edit/blog/nothing.md",
      "frontMatter": {
        "slug": "nothing",
        "title": "Nothing",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1539502055,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "permalink": "/blog/nothing",
      "prevItem": {
        "permalink": "/blog/author",
        "title": "Author",
      },
      "readingTime": 0.005,
      "source": "@site/blog/nothing.md",
      "tags": [],
      "title": "Nothing",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "both",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/both.md",
      "frontMatter": {
        "author": "ozaki",
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "author": "seb",
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "both",
        "title": "Both",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1609459200,
      "lastUpdatedBy": "seb",
      "nextItem": {
        "permalink": "/blog/lastUpdateDate",
        "title": "Last update date",
      },
      "permalink": "/blog/both",
      "prevItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "readingTime": 0.015,
      "source": "@site/blog/both.md",
      "tags": [],
      "title": "Both",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "lastUpdateDate",
    "metadata": {
      "authors": [],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/lastUpdateDate.md",
      "frontMatter": {
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "lastUpdateDate",
        "title": "Last update date",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1609459200,
      "lastUpdatedBy": undefined,
      "permalink": "/blog/lastUpdateDate",
      "prevItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "readingTime": 0.015,
      "source": "@site/blog/lastUpdateDate.md",
      "tags": [],
      "title": "Last update date",
      "unlisted": false,
    },
  },
]
`;

exports[`last update none 1`] = `
[
  {
    "content": "author",
    "id": "author",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "author",
      "editUrl": "https://baseEditUrl.com/edit/blog/author.md",
      "frontMatter": {
        "author": "ozaki",
        "last_update": {
          "author": "seb",
        },
        "slug": "author",
        "title": "Author",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "permalink": "/blog/author",
      "readingTime": 0.005,
      "source": "@site/blog/author.md",
      "tags": [],
      "title": "Author",
      "unlisted": false,
    },
  },
  {
    "content": "nothing",
    "id": "nothing",
    "metadata": {
      "authors": [],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "nothing",
      "editUrl": "https://baseEditUrl.com/edit/blog/nothing.md",
      "frontMatter": {
        "slug": "nothing",
        "title": "Nothing",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "permalink": "/blog/nothing",
      "prevItem": {
        "permalink": "/blog/author",
        "title": "Author",
      },
      "readingTime": 0.005,
      "source": "@site/blog/nothing.md",
      "tags": [],
      "title": "Nothing",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "both",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/both.md",
      "frontMatter": {
        "author": "ozaki",
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "author": "seb",
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "both",
        "title": "Both",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "nextItem": {
        "permalink": "/blog/lastUpdateDate",
        "title": "Last update date",
      },
      "permalink": "/blog/both",
      "prevItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "readingTime": 0.015,
      "source": "@site/blog/both.md",
      "tags": [],
      "title": "Both",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "lastUpdateDate",
    "metadata": {
      "authors": [],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/lastUpdateDate.md",
      "frontMatter": {
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "lastUpdateDate",
        "title": "Last update date",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": undefined,
      "permalink": "/blog/lastUpdateDate",
      "prevItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "readingTime": 0.015,
      "source": "@site/blog/lastUpdateDate.md",
      "tags": [],
      "title": "Last update date",
      "unlisted": false,
    },
  },
]
`;

exports[`last update time only 1`] = `
[
  {
    "content": "author",
    "id": "author",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "author",
      "editUrl": "https://baseEditUrl.com/edit/blog/author.md",
      "frontMatter": {
        "author": "ozaki",
        "last_update": {
          "author": "seb",
        },
        "slug": "author",
        "title": "Author",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": "seb",
      "nextItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "permalink": "/blog/author",
      "readingTime": 0.005,
      "source": "@site/blog/author.md",
      "tags": [],
      "title": "Author",
      "unlisted": false,
    },
  },
  {
    "content": "nothing",
    "id": "nothing",
    "metadata": {
      "authors": [],
      "date": 2024-03-07T17:28:43.000Z,
      "description": "nothing",
      "editUrl": "https://baseEditUrl.com/edit/blog/nothing.md",
      "frontMatter": {
        "slug": "nothing",
        "title": "Nothing",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": "Author",
      "nextItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "permalink": "/blog/nothing",
      "prevItem": {
        "permalink": "/blog/author",
        "title": "Author",
      },
      "readingTime": 0.005,
      "source": "@site/blog/nothing.md",
      "tags": [],
      "title": "Nothing",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "both",
    "metadata": {
      "authors": [
        {
          "imageURL": undefined,
          "name": "ozaki",
          "title": undefined,
          "url": undefined,
        },
      ],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/both.md",
      "frontMatter": {
        "author": "ozaki",
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "author": "seb",
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "both",
        "title": "Both",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": 1609459200,
      "lastUpdatedBy": "seb",
      "nextItem": {
        "permalink": "/blog/lastUpdateDate",
        "title": "Last update date",
      },
      "permalink": "/blog/both",
      "prevItem": {
        "permalink": "/blog/nothing",
        "title": "Nothing",
      },
      "readingTime": 0.015,
      "source": "@site/blog/both.md",
      "tags": [],
      "title": "Both",
      "unlisted": false,
    },
  },
  {
    "content": "last update date",
    "id": "lastUpdateDate",
    "metadata": {
      "authors": [],
      "date": 2020-01-01T00:00:00.000Z,
      "description": "last update date",
      "editUrl": "https://baseEditUrl.com/edit/blog/lastUpdateDate.md",
      "frontMatter": {
        "date": 2020-01-01T00:00:00.000Z,
        "last_update": {
          "date": 2021-01-01T00:00:00.000Z,
        },
        "slug": "lastUpdateDate",
        "title": "Last update date",
      },
      "hasTruncateMarker": false,
      "lastUpdatedAt": undefined,
      "lastUpdatedBy": "Author",
      "permalink": "/blog/lastUpdateDate",
      "prevItem": {
        "permalink": "/blog/both",
        "title": "Both",
      },
      "readingTime": 0.015,
      "source": "@site/blog/lastUpdateDate.md",
      "tags": [],
      "title": "Last update date",
      "unlisted": false,
    },
  },
]
`;
